src/Glpi/Search/Provider/SQLProvider.php

```diff
@@ -4751,7 +4751,7 @@ public static function constructData(array &$data, $onlycount = false)
                         "sql query fails (too many tables). " .
                         "Please use 'Items seen' criterion instead")
                 );
-            } else {
+            } else if ($_SESSION['glpi_use_mode'] === Session::DEBUG_MODE) {
                 echo $DBread->error();
```

Comment:
Using `trigger_error()` will fordward the error to the error handler, that will output the message only in dev/debug mode and that will also add some the error message in the `php-errors.log` file.

```suggestion
            } else {
                trigger_error($DBread->error(), E_USER_WARNING);
```

Reply:
Actually, it doesn't seem needed to show anything here since the DB error is already included in the SQL error message that is already logged to the log and output.